src/multiprogress.rs

```diff
@@ -48,6 +48,14 @@ impl MultiProgress {
         self.insert(bars_count, pb)
     }
 
+    /// Inserts a progress bar before the previous one and returns an internalized reference to it.
+    ///
+    /// The progress bar will be positioned before the previous progress bar in the [`MultiProgress`].
+    pub fn insert_before(&self, pb: ProgressBar) -> ProgressBar {
```

Comment:
This is, in fact, the `insert_before_the_last` method. I think exposing the `len` method and using `insert` in any combination is better.